src/lib/components/layout/Sidebar.svelte

```diff
@@ -477,7 +514,33 @@
 				</div>
 			{/if}
 
-			<div class="pl-2 my-2 flex-1 flex flex-col space-y-1 overflow-y-auto scrollbar-hidden">
+			<div
+				class="pl-2 my-2 flex-1 flex flex-col space-y-1 overflow-y-auto scrollbar-hidden"
+				on:scroll={async (e) => {
+					if (!$scrollPaginationEnabled) return;
+					if (tagView) return;
+					if (nextPageLoading) return;
+					if (chatPagniationComplete) return;
+
+					const maxScroll = e.target.scrollHeight - e.target.clientHeight;
+					const currentPos = e.target.scrollTop;
+					const ratio = currentPos / maxScroll;
+					if (ratio >= paginationScrollThreashold) {
+						nextPageLoading = true;
+						pageSkip.set($pageSkip + 1);
+						// extend existing chats
+						const nextPageChats = await getChatList(
+							localStorage.token,
+							$pageSkip * $pageLimit,
+							$pageLimit
+						);
+						// once the bottom of the list has been reached (no results) there is no need to continue querying
+						chatPagniationComplete = nextPageChats.length === 0;
+						await chats.set([...$chats, ...nextPageChats]);
+						nextPageLoading = false;
```

Comment:
could add a loading indicator to the bottom of the chat list. in case the user scrolls faster than their network can respond, it will inform the user that additional chats are still loading.